lib/model/rwfolder.go

```diff
@@ -1322,6 +1280,38 @@ func (f *rwFolder) performFinish(state *sharedPullerState) error {
 	return nil
 }
 
+func (f *rwFolder) handleOldFileOrOldDirectory(state *sharedPullerState) (err error) {
+	var fileInfo os.FileInfo
+	if fileInfo, err = osutil.Lstat(state.realName); err == nil {
```

Comment:
I'd invert the condition, so there is less indentation to consume.
```
err := os.Lstat....
if err != nil {
  return err
}

do rest
```

Also I still think it's easier for a human brain to process multiple ifs with early returns versus a nest of if else if else if